File: WdtcCore/src/main/java/org/wdt/wdtc/core/manger/VMManger.java
package org.wdt.wdtc.core.manger;

import lombok.Getter;
import org.wdt.utils.io.FileUtils;

import java.io.File;
import java.util.List;

public class VMManger {
  public static final String LAUNCHER_VERSION = "wdtc.launcher.version";
  public static final String CONFIG_PATH = "wdtc.config.path";
  public static final String DEBUG = "wdtc.debug.switch";
  public static final String APPLICATION_TYPE = "wtdc.application.type";
  public static final String CLIENT_ID = "wtdc.oauth.clientId";
  private static final List<String> LAUNCHER_AUTHOR = List.of("Wdt~");
  @Getter
  private static final String OS = System.getProperty("os.name");

  public static String getClientId() {
    return System.getProperty(CLIENT_ID, "8c4a5ce9-55b9-442e-9bd0-17cf89689dd0");
  }

  public static String getLauncherVersion() {
    return System.getProperty(LAUNCHER_VERSION, "demo");
  }

  public static boolean isDebug() {
    return Boolean.getBoolean(DEBUG);
  }

  public static File getWdtcConfigFromVM() {
    String WdtcConfigPath = System.getProperty(CONFIG_PATH);
    return WdtcConfigPath != null
        ? new File(FileUtils.getCanonicalPath(FileUtils.toFile(WdtcConfigPath)))
        : new File(System.getProperty("user.home"));
  }

  public static String getApplicationType() {
    return System.getProperty(APPLICATION_TYPE, "ui");
  }

  public static boolean isConsole() {
    return getApplicationType().equals("console");
  }

  public static boolean isUI() {
    return getApplicationType().equals("ui");
  }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/manger/VMManger.kt
package org.wdt.wdtc.core.manger

import java.io.File
import java.lang.Boolean.getBoolean

const val LAUNCHER_VERSION = "wdtc.launcher.version"
const val CONFIG_PATH = "wdtc.config.path"
const val DEBUG = "wdtc.debug.mode"
const val APPLICATION_TYPE = "wtdc.application.type"
const val CLIENT_ID = "wtdc.oauth.clientId"
const val NETWORK_TIMEOUT_TIME = "wdtc.net.timeout"
const val LOW_PERFORMANCE_MODE = "wdtc.low.performance.mode"

// TODO More and more people
private val LAUNCHER_AUTHOR = listOf("Wdt~")

val OS: String = System.getProperty("os.name")

val clientId: String = System.getProperty(CLIENT_ID, "8c4a5ce9-55b9-442e-9bd0-17cf89689dd0")

val launcherVersion: String = System.getProperty(LAUNCHER_VERSION, "demo")

val isDebug: Boolean = getBoolean(DEBUG)


val applicationType: String = System.getProperty(APPLICATION_TYPE, "ui")

val isConsole: Boolean = applicationType == "console"

val isUI: Boolean = applicationType == "ui"

val networkimeoutTime: Int = System.getProperty(NETWORK_TIMEOUT_TIME, "5000").toInt()

val isLowPerformanceMode = getBoolean(LOW_PERFORMANCE_MODE)

object VMManger {
	init {
		if (isDebug) System.setProperty(CONFIG_PATH, "./")
		if (System.getProperty(CONFIG_PATH) == null) {
			System.setProperty(CONFIG_PATH, System.getProperty("user.home"))
		}
	}
	
	val wdtcConfigFromVM: File
		get() {
			val wdtcConfigPath = System.getProperty(CONFIG_PATH)
			return if (wdtcConfigPath != null) File(File(wdtcConfigPath).canonicalPath)
			else File(System.getProperty("user.home"))
		}
	
}
